datamol/convert.py

```diff
@@ -164,8 +164,8 @@ def smiles_as_smarts(mol: Union[str, Chem.rdchem.Mol], keep_hs: bool = True) ->
     return smarts
 
 
-def to_inchi(mol: Union[str, Chem.rdchem.Mol]) -> Optional[str]:
-    """Convert a mol to Inchi.
+def to_inchi(mol: Union[str, dm.Mol]) -> Optional[str]:
```

Comment:
Hmm, i think taking the `MolToInchi` options and forwarding to rdkit here would be best here. 

Then the non standard can just call this one.  

On the non-standard one, I would say using a function with the appropriate name like `hash` `unique_id` etc would fit better.

Reply:
+1, and potentially precising some corner case limitations like Greg mentioned here https://github.com/rdkit/rdkit/discussions/4919#discussioncomment-2026036

Reply:
> Hmm, i think taking the MolToInchi options and forwarding to rdkit here would be best here.

I prefer preventing the users generating non-standard InChi using `to_inchi`. Duplicating the function to `to_inchi_non_standard` was done with this purpose in mind. Technically and API-wide, it does not change much, it's just a small code duplication without any complex logic.

On `unique_id`, I also prefer to stick to `to_inchi_non_standard` since it's more explicit. `unique_id` or `hash` means you have a solid way to create a unique ID or a unique hash for a given molecule. Unfortunately, uniqueness is not really well-defined for molecules as it's very context-dependent. See for example the decision from the InChi Trust to not include `FixedH` in the standard InChi (which is totally valid to me but also arbitrary on how you define "unique molecule").

That being said, we could have a "datamol branded" `unique_id` that under the hood call `to_inchikey_non_standard` Eventually we could also add another hash layer on top it.

Reply:
See the next commits and feel free to comment further if you have a strong feeling about this.